### PR TITLE
Adding TargetOS to the Container Name

### PR DESCRIPTION
**Pull request by venkat-raman251**
@jhendrixMSFT Please have a look.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets [hunk 1]
```diff
@@ -21,7 +21,8 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <ContainerName>$(TestProduct)-$(Branch)-$(BuildMoniker)</ContainerName>
+    <TargetOS>$(TargetQueue.Replace('.', '-'))</TargetOS>
+    <ContainerName>$(TestProduct)-$(Branch)-$(BuildMoniker)-$(TargetOS)</ContainerName>
```
Comment (jhendrixMSFT) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets]: Could you please add a condition to ContainerName that allows it to be overridden (see RunnerScript for an example)?

**Comment by venkat-raman251** [comment]
@jhendrixMSFT : Done.

File: src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets [hunk 2]
```diff
@@ -21,7 +21,8 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <ContainerName>$(TestProduct)-$(Branch)-$(BuildMoniker)</ContainerName>
+    <TargetOS>$(TargetQueue.Replace('.', '-'))</TargetOS>
```
Comment (jhendrixMSFT) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets]: This depends on the target queue having the same name as the target OS, which while true today might not be in the future.  It also adds drags in the full name of the queue which isn't really relevant (e.g. Windows.PerfInfraTest).  If we don't want to impose people putting the OS name in the BuildMoniker property I would rather add a dedicated TargetOS property to be passed explicitly.

Comment (venkat-raman251) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets]: Yep, agreed.

**Comment by venkat-raman251** [comment]
@jhendrixMSFT : Have just added TargetOS as a parameter that must be passed to the build

File: src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets [hunk 3]
```diff
@@ -21,7 +21,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <ContainerName>$(TestProduct)-$(Branch)-$(BuildMoniker)</ContainerName>
+    <ContainerName Condition="'$(ContainerName)' == ''">$(TestProduct)-$(Branch)-$(BuildMoniker)-$(TargetOS)</ContainerName>
```
Comment (jhendrixMSFT) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets]: You will also need to update target VerifyInputs to verify that the property was provided.

**Comment by jhendrixMSFT** [comment]
LGTM

**Pull request merged**